net: lib: nrf_cloud: P-GPS initialization improvements (#6840)
nrfconnect/sdk-nrf: doc/nrf/libraries/networking/nrf_cloud_pgps.rst

Ideally, once the device has connected to the cloud, the application must call the P-GPS initialization function.
If a connection is not available, initialization must still be called.

> If :kconfig:`CONFIG_NRF_CLOUD_PGPS_REQUEST_UPON_INIT` is enabled, the initialization function will not automatically download missing P-GPS data.
> In these cases, predictions might be unavailable until a connection is established to the cloud in the future.

Reviewer: ```suggestion
If the :kconfig:`CONFIG_NRF_CLOUD_PGPS_REQUEST_UPON_INIT` option is enabled, the initialization function does not automatically download missing P-GPS data.
In these cases, predictions might be unavailable until a connection is established to the cloud.
```

Author: That's fine, though some of that language was there before...
